lite/examples/pose_estimation/android/app/src/main/java/org/tensorflow/lite/examples/poseestimation/tracker/AbstractTracker.kt

```diff
@@ -0,0 +1,141 @@
+package org.tensorflow.lite.examples.poseestimation.tracker
+
+import org.tensorflow.lite.examples.poseestimation.data.Person
+
+abstract class AbstractTracker(val config: TrackerConfig) {
+
+    private val maxAge = config.maxAge * 1000 // convert milliseconds to microseconds
+    private var nextTrackId = 0
+    var tracks = mutableListOf<Track>()
+        private set
+
+    /**
+     * Computes pairwise similarity scores between detections and tracks, based
+     * on detected features.
+     * @param persons A list of detected person.
+     * @returns A list of shape [num_det, num_tracks] with pairwise
+     * similarity scores between detections and tracks.
+     */
+    abstract fun computeSimilarity(persons: List<Person>): List<List<Float>>
+
+    /**
+     * Tracks person instances across frames based on detections.
+     * @param persons A list of person
+     * @param timestamp The current timestamp in microseconds
+     * @return An updated list of persons with tracking id.
+     */
+    fun apply(persons: List<Person>, timestamp: Long): List<Person> {
+        tracks = filterOldTrack(timestamp).toMutableList()
+        val simMatrix = computeSimilarity(persons)
+        assignTrack(persons, simMatrix, timestamp)
+        tracks = updateTrack().toMutableList()
+        return persons
+    }
+
+    /**
+     * Clear all track in list of tracks
+     */
+    fun reset() {
+        tracks.clear()
+    }
+
+    /**
+     * Return the next track id
+     */
+    private fun nextTrackID() = ++nextTrackId
+
+    /**
+     * Performs a greedy optimization to link detections with tracks. The person
+     * list is updated in place by providing an `id` property. If incoming
+     * detections are not linked with existing tracks, new tracks will be created.
+     * @param persons A list of detected person. It's assumed that persons are
+     * sorted from most confident to least confident.
+     * @param simMatrix A list of shape [num_det, num_tracks] with pairwise
+     * similarity scores between detections and tracks.
+     * @param timestamp The current timestamp in microseconds.
+     */
+    private fun assignTrack(persons: List<Person>, simMatrix: List<List<Float>>, timestamp: Long) {
+        assert(simMatrix.size == persons.size && simMatrix[0].size == tracks.size) {
+            throw IllegalArgumentException()
```

Comment:
Better to output some error messages, such as printing out simMatrix.size, persons.size, simMatrix[0].size, tracks.size.